src/hooks/useSafeTimeout.ts

```diff
@@ -0,0 +1,33 @@
+import {useCallback, useEffect, useRef} from 'react'
+
+type SetTimeout = (handler: TimerHandler, timeout?: number, ...args: any[]) => number
+type ClearTimeout = (id: number) => void
+
+/**
+ * Safely call `setTimeout` and `clearTimeout` within a component.
+ *
+ * This hook ensures that all timeouts are cleared when the component unmounts.
+ */
+export default function useSafeTimeout(): {safeSetTimeout: SetTimeout; safeClearTimeout: ClearTimeout} {
+  const timers = useRef<Record<string, number>>({})
```

Comment:
I think using a `Set` here is a bit cleaner.

```
const timers = useRef<Set<number>>(new Set<number>())
// ...
timers.current.add(id)
// ...
clearTimeout(id)
timers.current.delete(id)
```

Reply:
👍